src/extensions/OrderBook.sol

```diff
@@ -0,0 +1,42 @@
+pragma solidity ^0.4.13;
+
+// NOTE: We're using address[100] to get back orderIDs from the orderFetcher as Solidity currently lacks
+// the ability to receive dynamic arrays from external contract calls. The list gets shrunk to the desired size
+// before returning
```

Comment:
I believe in this case we _can_ return a dynamic array because this function is not intended to be called by other contracts.  It is intended to be called by a client which (according to Solidity documentation) can receive a dynamic array result.  I believe all of the cases where we have this problem are in the same situation (called by client) so we can sidestep this whole issue.